<?xml version="1.0"?>
<!--
  起動コマンド： roslaunch telecoV dual_navigation.launch

  created by ©︎niscode
  2022年11月15日 ros-melodic版からコピー
-->

<launch>
  <arg name="rover_type"        default="mega3"  doc="mega3 / outdoor" />
  <arg name="lrf"               default="s2"     doc="s2"/>
  <arg name="node_start_delay"  default="5.0" />
  <arg name="start_rviz"        default="true" />

  <!-- bringup telecoV -->
  <include file="$(find telecoV)/launch/bringup_telecov.launch">
    <arg name="rover_type" value="$(arg rover_type)" />
    <arg name="lrf" value="$(arg lrf)" />
    <arg name="node_start_delay" value="$(arg node_start_delay)" />
  </include>

  <!-- オドメトリの配信ノード -->
  <node pkg="telecoV" type="pub_diff_odom" name="pub_diff_odom" />

  <!-- mapサーバー -->
  <!--以下でマップファイルのパスの指定 -->
  <arg name="map_file" default="$(find telecoV)/map/atr_1f.yaml"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

  <!-- 
    自己位置推定ノード amcl
    各パラメータについては http://wiki.ros.org/amcl
    を参照してください。
  -->
  <include file="$(find telecoV)/launch/amcl_diff.launch"/>
 
  <!--- 経路計画，指令値生成ノード　move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" args="/cmd_vel:=/rover_twist_base">
    <rosparam file="$(find telecoV)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find telecoV)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find telecoV)/params/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find telecoV)/params/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find telecoV)/params/dwa/dwa_local_planner_params_diff.yaml" command="load" />
    <rosparam file="$(find telecoV)/params/dwa/move_base_params.yaml" command="load" />
  </node>

  <!-- Rviz -->
  <arg name="rvizconfig" default="$(find telecoV)/rviz/navigation.rviz" />
  <node pkg="telecoV" type="pub_info" name="pub_info" />
  <!-- <node pkg="telecoV" type="multigoal_marker.py" name="multigoal_marker" /> -->
  <node pkg="robot_pose_publisher" type="robot_pose_publisher" name="robot_pose_publisher" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
  <group if="$(arg start_rviz)">
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rvizconfig)" />
  </group>
</launch>
